Build on Windows

jakirkham opened the pull request:
Fixes https://github.com/conda-forge/oniguruma-feedstock/issues/1

Reuses upstream's NMAKE-based build scripts to build this on Windows for 32-bit and 64-bit architectures and uses VS 2008 and VS 2015.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham commented:
Still needs tests on Windows.

jakirkham committed:
Drop skip on Windows

Looks like there are NMAKE files that can be used on Windows. So will
give that a try.

jakirkham committed:
Add Unix selectors on Unix specific tests

jakirkham committed:
Add Windows VC features

Provide standard VC features using Python to stitch them together to get
the build on Windows to produce packages for the different VCs.

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.4.5

Now that the Windows skip is dropped, re-render to enable builds on
AppVeyor and the CI badge in the README. Also make sure to add builds
for VS 2008 and VS 2015.

jakirkham committed:
Add Windows build script

Use CMake to build on Windows. There are NMAKE files as well, but some
of the build scripts for them are missing. So stick with CMake.

jakirkham commented:
Only built the shared libraries. Can build static libraries if there is demand later.

jakirkham committed:
Check for libraries and headers on Windows
